### PR TITLE
Create --service-account and --upload flags

### PR DESCRIPTION
**Pull request by fejta**
Default to not activating a service account and not uploading results.
Goal is to make it easier for people to run jobs outside of jenkins.

### CODE REVIEW

File: jenkins/bootstrap.py [hunk 1]
```diff
@@ -537,24 +536,34 @@ def build_name(started):
...
     # TODO(fejta): allow use of existing gcloud auth
+    if robot:
+        os.environ[SERVICE_ACCOUNT_ENV] = robot
+    if not os.getenv(SERVICE_ACCOUNT_ENV) and upload:
+        logging.warning('Cannot --upload=%s, no active gcloud account.', upload)
+        raise ValueError('--upload requires --service-account')
+    if not os.getenv(SERVICE_ACCOUNT_ENV) and not upload:
+        logging.info('Will not upload results.')
+        return
     if not os.path.isfile(os.environ[SERVICE_ACCOUNT_ENV]):
```
Comment (ixdy) [inline jenkins/bootstrap.py]: isn't this always  is going to always fail if `robot` is `None`? (unless users explicitly set `GOOGLE_APPLICATION_CREDENTIALS` in the environment... but then I don't understand why we'd need `--service-account`.)

Comment (ixdy) [inline jenkins/bootstrap.py]: oh, the previous conditional will drop us out.

**Comment by mikedanese** [comment]
This is breakin bazel presubmit.

**Pull request merged**